src/azul/service/hca_response_v5.py

```diff
@@ -298,9 +311,73 @@ def make_projects(self, entry):
                 translated_project['insdcProjectAccessions'] = project.get('insdc_project_accessions', [None])
                 translated_project['insdcStudyAccessions'] = project.get('insdc_study_accessions', [None])
                 translated_project['supplementaryLinks'] = project.get('supplementary_links', [None])
+                translated_project['contributorMatrices'] = self.make_contributor_matrices(entry)
             projects.append(translated_project)
         return projects
 
+    # FIXME: Move this to during aggregation
+    #        https://github.com/DataBiosphere/azul/issues/2415
+
+    # FIXME: More test coverage of this method code in isolation
+    #        https://github.com/DataBiosphere/azul/issues/2416
+
+    def make_contributor_matrices(self, entry) -> JSON:
+        """
+        Returns a stratification tree for the contributor-generated matrix files
+        in the entry.
+        """
+        matrices = entry['contents']['contributor_matrices']
+        if matrices:
+            files = one(matrices)['file']
+        else:
+            files = []
+
```

Comment:
Add assertion here to ensure uniqueness of file uuid and file name. File name needs to be added to the indexer.